src/proxy.rs

```diff
@@ -17,6 +19,13 @@ use tower_http::trace::TraceLayer;
 
 use crate::serve::ServerResult;
 
+/// The `X-Forwarded-Host`` (XFH) header is a de-facto standard header for
+/// identifying the original host requested by the client in the Host HTTP
+/// request header.
+///
+/// Refer: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
+const X_FORWARDED_HOST: &str = "x-forwarded-host";
+
```

Comment:
Turns out `HeaderName` has interior mutability!

```log
warning: a `const` item should never be interior mutable
  --> src/proxy.rs:27:1
   |
27 | const X_FORWARDED_HOST: HeaderName = HeaderName::from_static("x-forwarded-host");
   | -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | make this a static item (maybe with lazy_static)
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#declare_interior_mutable_const
   = note: `#[warn(clippy::declare_interior_mutable_const)]` on by default
```

So I had to leave it as a `&str`

Reply:
Yes, I do recall that mess. To my understanding it's actually not, but uses something that could be. Using a lazy static should work though.

Reply:
I think its okay with current approach? What do you think? Using lazy_static just for this constant feels a bit overkill imho.